packages/cache/__tests__/hydration.js

```diff
@@ -35,3 +35,145 @@ test('rehydrated styles to head can be flushed', () => {
   cache.sheet.flush()
   expect(document.documentElement).toMatchSnapshot()
 })
+
+test('flushing rehydrated styles in the head only affect styles matching the cache key', () => {
```

Comment:
I'm wondering - was this supposed to test a fix or is it just a new test for something that was not tested up until now?

Reply:
Hey 👋 yeah it was a bit of both, without the fix  for filtering the cache key (`data-emotion="emo"`), both style tags would be removed. AFAIK that's why we were having issues using the CacheProvider before

Reply:
This test doesn't test any changed behavior, the styles in this test are both from server-rendering which was already handled correctly.